src/main/java/org/openmicroscopy/ms/zarr/Configuration.java

```diff
@@ -48,6 +48,34 @@
     private String netPath = getRegexForNetPath("/image/" + PLACEHOLDER_IMAGE_ID + ".zarr/");
     private int netPort = 8080;
 
+    /* Note a singleton configuration for subsequent injection by Spring. */
+
+    private static Configuration SINGLETON_CONFIGURATION;
```

Comment:
Why is a singleton needed?

Reply:
Three different beans (so far) need to consult it so I figured nicer to have Spring provide it rather than having to feed it through from one class to another in the Java code, then at a glance at the XML one can see which beans are consuming it. I figured better to use this approach of priming a factory method to avoid spending time figuring how best to refactor the argv-handling stuff. Making the configuration itself non-static feels easier to get right in setup for unit tests that need specific settings.

Reply:
Hmm.... not sure I follow. Should the use of dependency injection prevent the need for a singleton? I think of DI as the solution to static state.

Reply:
Yes, this is the quick hack around that: (a) I couldn't quickly see how to get `argv` into a Spring-instantiated configuration instance (did I miss something in `System`?) and (b) I couldn't easily get Spring to be happy with my registering an instantiated configuration bean in the application context before launching it. At least, without having to make configuration instances writable. So, like the lucky turn toward the end of THX-1138, I figured I was in danger of spending too much time on that aspect and I figured this increment at least gets us as far as DI from the consumer view until I have an inspiring dream about the supply side.

Reply:
Fair enough. Happy to live with it as a hack.

Reply:
> until I have an inspiring dream

I got some more sleep, that seems to have helped, heh.